linera-views/tests/operations_tests.rs

```diff
@@ -205,103 +208,180 @@ async fn test_readings_memory_specific() {
     run_readings_vec(key_value_store, key_value_vec).await;
 }
 
-#[cfg(test)]
-async fn run_writings_random<OP: KeyValueStoreClient + Sync>(key_value_store: OP) {
-    let mut rng = rand::rngs::StdRng::seed_from_u64(2);
-    let mut kv_state = BTreeMap::new();
-    let n_oper = 100;
-    let size_batch = 8;
-    // key space has size 4^4 = 256 so we necessarily encounter collisions
-    // because the number of generated keys is about size_batch * n_oper = 800 > 256.
-    let key_prefix = vec![0];
-    for _ in 0..n_oper {
+fn generate_random_batch<R: RngCore>(
+    rng: &mut R,
+    key_prefix: &[u8],
+    size_batch: usize,
+    option: usize,
+) -> Batch {
+    if option == 1 {
         let mut batch = Batch::new();
+        // Fully random batch
         for _ in 0..size_batch {
             let choice = rng.gen_range(0..8);
             // Inserting a key
-            if choice < 5 {
+            if choice < 6 {
                 // Insert
-                let key = get_small_key_space(&mut rng, &key_prefix, 4);
+                let key = get_small_key_space(rng, key_prefix, 4);
                 let len_value = rng.gen_range(0..10); // Could need to be split
-                let value = get_random_byte_vector(&mut rng, &[], len_value);
+                let value = get_random_byte_vector(rng, &[], len_value);
                 batch.put_key_value_bytes(key.clone(), value.clone());
-                kv_state.insert(key, value);
             }
             if choice == 6 {
                 // key might be missing, no matter, it has to work
-                let key = get_small_key_space(&mut rng, &key_prefix, 4);
-                kv_state.remove(&key);
+                let key = get_small_key_space(rng, key_prefix, 4);
                 batch.delete_key(key);
             }
             if choice == 7 {
                 let len = rng.gen_range(1..4); // We want a non-trivial range
-                let delete_key_prefix = get_small_key_space(&mut rng, &key_prefix, len);
+                let delete_key_prefix = get_small_key_space(rng, key_prefix, len);
                 batch.delete_key_prefix(delete_key_prefix.clone());
+            }
+        }
+    }
+    if option == 2 {
+        // A renormalized batch.
+        let batch = generate_random_batch(rng, key_prefix, size_batch, 1);
+        let unord_batch = batch.simplify();
+        return unord_batch.as_batch();
+    }
+    Batch::new()
+}
+
+fn get_key(key_prefix: &[u8], key_suffix: Vec<u8>) -> Vec<u8> {
+    let mut key = key_prefix.to_vec();
+    key.extend(key_suffix);
+    key
+}
+
+fn generate_specific_batch(key_prefix: &[u8], option: usize) -> Batch {
```

Comment:
```suggestion
fn generate_specific_batch(key_prefix: &[u8], multiple_entries: bool) -> Batch {
```

Or `put_and_delete`?

Reply:
Looks like we want an enum?